web/src/attachments_ui.ts

```diff
@@ -223,3 +224,60 @@ export function set_up_attachments(): void {
         },
     });
 }
+
+export function suggest_delete_detached_attachments(attachments_list: ServerAttachment[]): void {
+    const html_body = render_confirm_delete_detached_attachments_modal({
+        attachments_list,
+        realm_allow_edit_history: realm.realm_allow_edit_history,
+    });
+
+    // Since we want to delete multiple attachments, we want to be
+    // able to keep track of attachments to delete and which ones to
+    // retry if it fails. A list is way too slow for the current use
+    // case. By Build a map from the list of attachments we want to
+    // delete, we can perform searches and deletion at O(1) time.
+    const attachments_map = new Map<number, ServerAttachment>();
+    for (const attachment of attachments_list) {
+        attachments_map.set(attachment.id, attachment);
+    }
+
+    function do_delete_attachments(): void {
+        dialog_widget.show_dialog_spinner();
+        for (const [key, attachment] of attachments_map.entries()) {
+            const id = Number(key);
+            void channel.del({
+                url: "/json/attachments/" + attachment.id,
+                success() {
+                    attachments_map.delete(id);
+                    if (attachments_map.size === 0) {
+                        dialog_widget.hide_dialog_spinner();
+                        dialog_widget.close();
+                    }
+                },
+                error() {
+                    dialog_widget.hide_dialog_spinner();
+                    ui_report.error(
+                        $t_html({defaultMessage: "One or more files could not be deleted."}),
```

Comment:
Would it make sense to just display the error that the server served? That feels more informative than this generic error, since it'd tell the user if it was a permissions problem or something else.

I'm just not sure if we need an extra translated string for this rare situation.

Reply:
I think it might be better if we provide a better message witht the banner? I'm not sure what the server serves. We could start a CZO discussion on this as a follow-up.

Reply:
Sure, please start that conversation; you should be able to test it pretty easily by just parsing the `xhr` returned by the server in this case.